internal/mempool/mempool.go

```diff
@@ -363,6 +363,7 @@ func (txmp *TxMempool) CheckTx(
 	if err != nil {
 		removeHandler(true)
 		res.Log = txmp.AppendCheckTxErr(res.Log, err.Error())
+		return err
```

Comment:
This would break the existing logic if we return err instead of putting err in the response. Currently error is already being handled when adding to the res.log and passing back to cb function

Reply:
After a second though, I think it makes more sense to return error to the client rather than putting error in the response log, so the change LGTM. Probably need to evaluate whether there's any risk/impact to the existing client or not.